packages/core/src/main/Account.ts

```diff
@@ -364,12 +382,22 @@ export class Account extends EventEmitter {
     if (!this.service) {
       throw new Error('Services are not set.');
     }
+    this.logger.info(`Creating new client {mls: ${!!this.enableMLS}}`);
     const registeredClient = await this.service.client.register(loginData, clientInfo, entropyData);
+    if (this.enableMLS) {
+      this.coreCryptoClient = await this.createMLSClient(registeredClient);
+      await this.service.client.uploadMLSPublicKeys(this.coreCryptoClient.clientPublicKey(), registeredClient.id);
+      await this.service.client.uploadMLSKeyPackages(
+        this.coreCryptoClient.clientKeypackages(this.nbPrekeys),
+        registeredClient.id,
+      );
```

Comment:
isn't it better to wrap this part in a try-catch and have a nice error message for it?